libimage/filters.go

```diff
@@ -16,11 +16,11 @@ import (
 
 // filterFunc is a prototype for a positive image filter.  Returning `true`
 // indicates that the image matches the criteria.
-type filterFunc func(*Image) (bool, error)
+type filterFunc func(*Image, []string) (bool, error)
 
 // filterImages returns a slice of images which are passing all specified
 // filters.
-func filterImages(images []*Image, filters []filterFunc) ([]*Image, error) {
+func filterImages(images []*Image, filters []filterFunc, registries []string) ([]*Image, error) {
```

Comment:
registries should be of type `[]c/image/docker/reference.Reference`.

Reply:
Why?  	registries, err := sysregistriesv2.UnqualifiedSearchRegistries(r.systemContextCopy())
Returns a []string

Seems like we would need to convert this to string to a reference only later to be converted back to a string.

Reply:
We need it to use https://github.com/containers/image/blob/main/docker/reference/helpers.go#L36